app/Actions/Photo/Strategies/RotateStrategy.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace App\Actions\Photo\Strategies;
+
+use App\Actions\Photo\Extensions\SourceFileInfo;
+use App\Contracts\SizeVariantFactory;
+use App\Contracts\SizeVariantNamingStrategy;
+use App\Facades\Helpers;
+use App\Image\ImageHandlerInterface;
+use App\Metadata\Extractor;
+use App\Models\Logs;
+use App\Models\Photo;
+use App\Models\SizeVariant;
+
+class RotateStrategy extends AddBaseStrategy
+{
+	protected int $direction;
+
+	public function __construct(Photo $photo, int $direction)
+	{
+		// We exploit the "add strategy" here, because rotation of a photo
+		// has a lot in common with adding a new photo.
+		// We first make a temporary copy of the photo, rotate that copy
+		// and then "re-import" that temporary copy into an existing photo
+		// model.
+		// As we want the temporary file to be moved back into place,
+		// we delete the "imported" file and do not want to import via symlink.
+		// We do not want to skip duplicates (in case the photo is already a
+		// duplicate, we still want to rotate it) and we want to re-sync
+		// metadata (after rotation width, height and filesize may have changed).
+		//
+		// TODO: Think again about the following potentially, undesired behaviour:
+		//
+		// In case the photo has originally been imported as an symbolic link,
+		// the photo won't be a symbolic link after rotation, but become an
+		// independent file which is detached from the original target of the
+		// symbolic link.
+		parent::__construct(
```

Comment:
Personally I consider this _undesired_ behaviour as _desired_.
**Lychee MUST NOT modify originals imported via symbolic link.**

At least that is my take on the expectation from import via sym link.

Reply:
Yep, I originally coded it that way deliberately. Basically, I had two choices. Either bail out with an error message if the user attempts to rotate a photo that was imported via a symlink, or replace the symlink with an actual file with the rotated contents. I thought the former was needlessly restrictive so I went for the latter. Well, I suppose there was a third choice: modify the file pointed to by the symlink, but I think users would revolt if we did that :wink:.

Reply:
OK, I will add a comment to the code which documents this design decision.